server/routerlicious/packages/services-telemetry/src/lumber.ts

```diff
@@ -0,0 +1,141 @@
+        }
+        return this._durationInMs;
+    }
+
+    public get successful(): boolean | undefined {
+        if (this.type === LumberType.Log) {
+            return undefined;
+        }
+        return this._successful;
+    }
+
+    public get message(): string | undefined {
+        return this._message;
+    }
+
+    public get statusCode(): string | undefined {
+        return this._statusCode;
+    }
+
+    public get exception(): Error | undefined {
+        return this._exception;
+    }
+
+    public get logLevel(): LogLevel | undefined {
+        return this._logLevel;
+    }
+
+    constructor(
+        public readonly eventName: T,
+        public readonly type: LumberType,
+        private readonly engineList: ILumberjackEngine[],
+        properties?: Map<string, any> | Record<string, any>) {
+            if (properties) {
+                this.addProperties(properties);
+            }
+        }
+
+    public addProperty(key: string, value: any): this {
+        this._properties.set(key, value);
+        return this;
+    }
+
+    public addProperties(properties: Map<string, any> | Record<string, any>): this {
+        if (properties instanceof Map) {
+            if (this._properties.size === 0) {
+                this._properties = properties;
+            } else {
+                properties.forEach((value: any, key: string) => { this.addProperty(key, value); });
+            }
+        } else {
+            Object.entries(properties).forEach((entry) => {
+                const [key, value] = entry;
+                this.addProperty(key,value);
+            });
+        }
+        return this;
+    }
+
+    public success(
+        message: string,
+        statusCode: number | string | undefined,
+        metadata: ITelemetryMetadata,
+        logLevel: LogLevel = LogLevel.Info) {
+        this.emit(message, statusCode, metadata, logLevel, true);
+    }
+
+    public error(
+        message: string,
+        statusCode: number | string | undefined,
+        metadata: ITelemetryMetadata,
+        exception?: Error | undefined,
+        logLevel: LogLevel = LogLevel.Error) {
+        this.emit(message, statusCode, metadata, logLevel, false, exception);
+    }
+
+    private emit(
+        message: string,
+        statusCode: number | string | undefined,
+        metadata: ITelemetryMetadata,
+        logLevel: LogLevel,
+        successful: boolean,
+        exception?: Error) {
+        if (this.completed) {
+            throw new Error(
+                `Trying to complete a Lumber telemetry operation ${this.eventName} that has alredy been completed.`);
+        }
+
+        this._message = message;
+        if (statusCode) {
+            this._statusCode = statusCode.toString();
+        }
+        this._metadata = metadata;
+        this._logLevel = logLevel;
+        this._successful = successful;
+        this._exception = exception;
+        this._durationInMs = Date.now() - this.startTime;
```

Comment:
Should we use [performance.now()](https://nodejs.org/api/perf_hooks.html#perf_hooks_performance_now) instead of `Date.now()` for better precision, similar to [PerformanceEvent in the client](https://github.com/microsoft/FluidFramework/blob/main/packages/utils/telemetry-utils/src/logger.ts#L405)?